src/UI/Implementation/Component/Button/Toggle.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/* Copyright (c) 2018 Thomas Famula <famula@leifos.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Implementation\Component\Button;
+
+use ILIAS\UI\Component as C;
+use ILIAS\UI\Implementation\Component\ComponentHelper;
+use ILIAS\UI\Implementation\Component\JavaScriptBindable;
+use ILIAS\UI\Implementation\Component\Triggerer;
+use ILIAS\UI\Implementation\Component\TriggeredSignal;
+
+class Toggle extends Button implements C\Button\Toggle {
+	use ComponentHelper;
+	use JavaScriptBindable;
+	use Triggerer;
+
+	/**
+	 * @var string
+	 */
+	protected $label;
+
+	/**
+	 * @var bool
+	 */
+	protected $is_on;
+
+	/**
+	 * @var string|null
+	 */
+	protected $action_off;
+
+	protected $action_off_signal;
+	protected $action_on_signal;
+
+	public function __construct($label, $action, $action_off, $is_on)
+	{
+		$this->checkStringArg("label", $label);
+		$this->checkStringOrSignalArg("action", $action);
+		$this->checkStringOrSignalArg("action_off", $action_off);
+		$this->checkBoolArg("is_on", $is_on);
+
+		$this->label = $label;
+
+		if (is_string($action)) {
+			$this->action = $action;
+		}
+		else {
+			$this->action = null;
+			$this->action_on_signal = new TriggeredSignal($action, "click");
+		}
+
+		if (is_string($action_off)) {
+			$this->action_off = $action_off;
+		}
+		else {
+			$this->action_off = null;
+			$this->action_off_signal = new TriggeredSignal($action_off, "click");
+		}
+
+		$this->is_on = $is_on;
+	}
+
+	public function getLabel()
+	{
+		return $this->label;
+	}
+
+	/**
+	 * @inheritdoc
+	 */
+	public function isOn()
+	{
+		return $this->is_on;
+	}
+
+	/**
+	 * @inheritdoc
+	 */
+	public function getActionOff()
+	{
+		if ($this->action_off !== null) {
+			return $this->action_off;
+		}
+		return $this->action_off_signal;
```

Comment:
I think this makes the typehint on `getActionOff`-incorrect. `$action_off_signal` seems to be a `TriggeredSignal`, not a `Signal[]` or a `string`. Same goes for `getAction`.

Reply:
What I do not understand is how the interface depends on the implementation here. E.g. the interface Component\Triggerer is using implementation namespaces:
use ILIAS\UI\Implementation\Component\TriggeredSignal;

Reply:
> The interface Component\Triggerer is using implementation namespaces:

This is indeed correct. IMO this is not disallowed (at least I don't know where), but unaesthetic at least. I also think this goes beyond the scope of this PR.